force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,48 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+  if (trigger.isBefore && trigger.isUpdate) {
+    for (opportunity opp : Trigger.new) {
+      if (opp.Amount < 5000) {
+          opp.addError('Opportunity amount must be greater than 5000');   
+      }
+    }
+  }
+         if (Trigger.isBefore && Trigger.isDelete) {
+          Set <Id> accountIds = new Set<Id>();
+          for (Opportunity opp : Trigger.old) {
+             accountIds.add(opp.AccountId);
+          }
+          Map <Id, Account> accountsWithBanking = new Map<Id, Account>();
+          List<Account> accountsForOpps = [SELECT Id, Industry FROM Account WHERE Id IN :accountIds];
+          for (Account acc : accountsForOpps) {
+            if (acc.Industry == 'Banking') {
```

Comment:
You could just filter this Industry in your query and then you would pull less data and not need the extra check here.